zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/publishing/Publisher.java

```diff
@@ -787,7 +787,7 @@ private static void sendToKafka(Collection collection) throws IOException {
     // Valid CMDDataset uris for published CMD versions of a dataset (edition) - /dataset/{datatsetId}/editions/{edition}/versions/{version}/metadata
     protected static boolean isValidCMDDatasetURI (String uri){
 
-        return uri.chars().filter(ch -> ch == '/').count() > 6;
+        return uri.chars().filter(ch -> ch == '/').count() > 5;
```

Comment:
I think this is potentially too open to be able accurately determined if a value is a CMD uri. Also not sure of the specific context here but technically a `/datasets` is a valid CMD uri.

From the example in the test class I think this is checking a dataset version uri? If so I think a regex would be a better approach as there is a specific uri pattern for this:

`/datasets/{datasetID}/editions/{edition}/versions/{version}`

I think it should be reasonably straight forward to implement this checking if the input string starts with that pattern. I think this would be a more robust solution.

Reply:
I found this as simple and single line change with cleaner and more readable solution using core library. I have not tried but regex uses Pattern and Matcher possibly involving more line of code changes.

Reply:
I appreciate you're updating the code that already existed but for the reasons stated above this should probably change as IMO it doesn't provide a rigorous enough validation to correctly reject values that don't meet the criteria which I why I think while we are here we should update it.

In terms of scope creep - if this were multiple methods/classes then we'd have to make judgement on cost vs return but in this case it's 1 method so the impact will be limited and it shouldn't take too long to implement.

Reply:
Done, regex pattern applied